### PR TITLE
feat: account for Japanese whitespace

### PR DESCRIPTION
**Pull request by fosskers**
Japanese whitespace... the silent killer.

### CODE REVIEW

File: test/test-str.lisp [hunk 1]
```diff
@@ -172,7 +172,8 @@
 (test trim
   (is (string= "rst" (trim "  rst  ")))
   (is (string= nil (trim nil)))
-  (is (string= "rst" (trim "drste" :char-bag "de"))))
+  (is (string= "rst" (trim "drste" :char-bag "de")))
+  (is (string= "rst" (trim "　　rst　　　"))))
```
Comment (fosskers) [inline test/test-str.lisp:176]: This renders on Github as normal-looking spaces, but they aren't. Emacs renders them clearly with a cyan underline.

**Comment by kilianmh** [comment]
After the disaster last year, we should test on all implementations before adding characters.
For me loading works with `sbcl`, `ccl`, `ecl`, `cmu`, and `clisp`.
On abcl during loading `str` we get:
```common-lisp
Unrecognized character name: "Ideographic_space"
   [Condition of type ERROR]
```
I guess we have to work with the character-code on `abcl` at least and maybe also other implementations.

**Comment by fosskers** [comment]
~~I will test on Clasp.~~

Tests pass on Clasp.

File: str.lisp [hunk 2]
```diff
@@ -135,6 +135,7 @@
 (defvar *whitespaces* (list #\Backspace #\Tab #\Linefeed #\Newline #\Vt #\Page
                             #\Return #\Space #\Rubout
                             #+sbcl #\Next-Line #-sbcl (code-char 133)
+                            #-abcl #\Ideographic_space
```
Comment (fosskers) [inline str.lisp]: Or, is it possible to use `code-char` like the SBCL line is doing and specify the code manually?

**Comment by kilianmh** [comment]
Abcl now works! +1
So remaining are at least GCL, lispworks and allegro.

**Comment by kilianmh** [comment]
Maybe @CK-DE you can check with lispworks?

**Comment by kilianmh** [comment]
allegro 11 works as well for me

**Comment by fosskers** [comment]
I'd test on GCL but I [can't for the life of me get it to compile.](https://savannah.gnu.org/bugs/?65905)

**Comment by kilianmh** [comment]
I just tested and both `lispworks` and `gcl` both error on `#\Ideographic_space` so should also use only the char-code. After that it should be ready to merge ;)

**Comment by fosskers** [comment]
Ah nice, those are two I can't test. I'll add the feature for those two tomorrow morning.

**Review by kilianmh — APPROVED** [review]

**Comment by kilianmh** [comment]
Thank you @fosskers for improving this library!

**Pull request merged**